presto-verifier/src/main/java/com/facebook/presto/verifier/rewrite/FunctionCallRewriter.java

```diff
@@ -0,0 +1,136 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.facebook.presto.verifier.rewrite;
+
+import com.facebook.presto.sql.tree.Expression;
+import com.facebook.presto.sql.tree.ExpressionRewriter;
+import com.facebook.presto.sql.tree.ExpressionTreeRewriter;
+import com.facebook.presto.sql.tree.FunctionCall;
+import com.facebook.presto.sql.tree.Node;
+import com.facebook.presto.sql.tree.QualifiedName;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+
+import java.util.List;
+import java.util.Map;
+import java.util.Objects;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+import java.util.stream.Collectors;
+
+import static com.google.common.collect.ImmutableList.toImmutableList;
+import static java.util.Objects.requireNonNull;
+
+public class FunctionCallRewriter
+        extends DefaultTreeRewriter<Void>
+{
+    public static class FunctionCallSubstitute
+    {
+        private final QualifiedName name;
+        private final List<Integer> originalArgumentIndices;
+
+        public FunctionCallSubstitute(QualifiedName name, List<Integer> originalArgumentIndices)
+        {
+            this.name = requireNonNull(name, "name is null");
+            this.originalArgumentIndices = ImmutableList.copyOf(requireNonNull(originalArgumentIndices, "originalArgumentIndices is null"));
+        }
+
+        @Override
+        public boolean equals(Object o)
+        {
+            if (this == o) {
+                return true;
+            }
+            if (o == null || getClass() != o.getClass()) {
+                return false;
+            }
+            FunctionCallSubstitute that = (FunctionCallSubstitute) o;
+            return name.equals(that.name) && Objects.equals(originalArgumentIndices, that.originalArgumentIndices);
+        }
+
+        @Override
+        public int hashCode()
+        {
+            return Objects.hash(name, originalArgumentIndices);
+        }
+    }
+
+    private static final Pattern FUNCTION_CALL_SUBSTITUTION_PATTERN = Pattern.compile("/(\\w+)\\(([\\w|,]+)\\)/(\\w+)\\(([\\w|,]+)\\)/");
```

Comment:
Can you add a comment on what this regex matches ?

Reply:
+1
And one or two examples. It takes me a significant effort every time to understand any kind of regex - I don't work with them too often.

Reply:
Split the pattern definition, and hope it makes it a bit easier to read.